expression/builtin_compare.go

```diff
@@ -469,7 +474,7 @@ func (c *greatestFunctionClass) getFunction(ctx sessionctx.Context, args []Expre
 	case types.ETDecimal:
 		sig = &builtinGreatestDecimalSig{bf}
 		sig.setPbCode(tipb.ScalarFuncSig_GreatestDecimal)
-	case types.ETString:
+	case types.ETString, types.ETDuration:
```

Comment:
tp never be types.ETDuration, because tp will be set to types.ETString in L451?

Reply:
Done.